File: backend/src/main/java/org/example/financemanager/domain/Transaction.java
package org.example.financemanager.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "transaction", schema = "public")
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Getter
@Setter
@Builder
public class Transaction {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "description")
    private String description;

    @Column(name = "date")
    private LocalDate date;

    @Column(name = "type")
    @Enumerated(EnumType.STRING)
    private TransactionTypes type;

    @Column(name = "amount")
    private Double amount;

    @Column(name = "currency")
    @Enumerated(EnumType.STRING)
    private Currency currency;

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "category_id", referencedColumnName = "id")
    private Category category;
}

File: frontend/src/app/common/transaction.ts
import {CategoryDto} from "./category-dto";

export class Transaction {
  public id: number;
  public description: string;
  public date: string;
  public type: string;
  public amount: number;
  public currency: string;
  public category: CategoryDto;
  constructor(id: number, description: string, date: string, type: string, amount: number, currency: string, category: CategoryDto) {
    this.id = id;
    this.description = description;
    this.date = date;
    this.type = type;
    this.amount = amount;
    this.currency = currency;
    this.category = category;
  }
}
